src/rad/appdata.cpp

```diff
@@ -1953,7 +1953,192 @@ void ApplicationData::ConvertObject( ticpp::Element* parent, int fileMajor, int
 			}
 		}
 	}
+
 	/* The file is now at least version 1.12 */
+	// TODO: Dont know where Version 1.13 comes from, so this is for Version 1.14
+	if (fileMajor < 1 || (1 == fileMajor && fileMinor < 14)) {
+		// Rename all wx*_BORDER-Styles to wxBORDER_*-Styles and remove wxDOUBLE_BORDER
+		oldProps.clear();
+		newProps.clear();
+		oldProps.insert("style");
+		oldProps.insert("window_style");
+		GetPropertiesToConvert(parent, oldProps, &newProps);
+
+		for (newProp = newProps.begin(); newProp != newProps.end(); ++newProp) {
+			wxString styles = _WXSTR((*newProp)->GetText(false));
+			if (!styles.empty()) {
+				if (TypeConv::FlagSet(wxT("wxSIMPLE_BORDER"), styles)) {
+					styles = TypeConv::ClearFlag(wxT("wxSIMPLE_BORDER"), styles);
+					styles = TypeConv::SetFlag(wxT("wxBORDER_SIMPLE"), styles);
+				}
+
+				if (TypeConv::FlagSet(wxT("wxDOUBLE_BORDER"), styles)) {
+					styles = TypeConv::ClearFlag(wxT("wxDOUBLE_BORDER"), styles);
+				}
+
+				if (TypeConv::FlagSet(wxT("wxSUNKEN_BORDER"), styles)) {
+					styles = TypeConv::ClearFlag(wxT("wxSUNKEN_BORDER"), styles);
+					styles = TypeConv::SetFlag(wxT("wxBORDER_SUNKEN"), styles);
+				}
+
+				if (TypeConv::FlagSet(wxT("wxRAISED_BORDER"), styles)) {
+					styles = TypeConv::ClearFlag(wxT("wxRAISED_BORDER"), styles);
+					styles = TypeConv::SetFlag(wxT("wxBORDER_RAISED"), styles);
+				}
+
+				if (TypeConv::FlagSet(wxT("wxSTATIC_BORDER"), styles)) {
+					styles = TypeConv::ClearFlag(wxT("wxSTATIC_BORDER"), styles);
+					styles = TypeConv::SetFlag(wxT("wxBORDER_STATIC"), styles);
+				}
+
+				if (TypeConv::FlagSet(wxT("wxNO_BORDER"), styles)) {
+					styles = TypeConv::ClearFlag(wxT("wxNO_BORDER"), styles);
+					styles = TypeConv::SetFlag(wxT("wxBORDER_NONE"), styles);
+				}
+
+				(*newProp)->SetText(_STDSTR(styles));
+			}
+		}
+
+		// wxBitmapButton: Remove wxBU_AUTODRAW and rename properties selected->pressed,
```

Comment:
I've found that `wxBU_AUTODRAW` is marked obsolete in wxWidgets' source. Is there any documentation on this?

Why does the `wxBitmapButton` constructor still pass it as a style? http://docs.wxwidgets.org/3.1/classwx_bitmap_button.html

Reply:
``wxBitmapButton`` itself is obsolete. I think that style got simply removed from the docs when they did their ``wxAnyButton`` refactoring.